core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt

```diff
@@ -121,52 +124,51 @@ class UnitMovementAlgorithms(val unit:MapUnit) {
                     path.add(intermediateTile)
                     intermediateTile = movementTreeParents[intermediateTile]!!
                 }
-                return path.reversed() // and reverse in order to get the list in chronological order
+                path.reverse() // and reverse in order to get the list in chronological order
```

Comment:
`reverse()` is better since it reverses in place rather than reverses into a new copy of the list, which is what `reversed()` does